src/core/server/saved_objects/migrations/core/document_migrator.ts

```diff
@@ -680,8 +699,15 @@ function wrapWithTry(
       const failedDoc = JSON.stringify(doc);
       log.error(error);
 
-      throw new Error(
-        `Failed to transform document ${doc?.id}. Transform: ${failedTransform}\nDoc: ${failedDoc}`
+      // changing the error being thrown to an object or an instance of Either.left requires refactoring a lot of try/catch implementations further upstream within v2 migrations.
+      // as an initial improvement, we're adding more information to the error that's thrown (full error itself).
+      throw new TransformSavedObjectDocumentError(
+        doc.id,
+        doc.type,
+        doc.namespace ?? 'N/A',
+        failedTransform,
```

Comment:
Thanks for pointing that out and the suggested change!

Reply:
If the only usage of `doc.namespace` is to generate the technical id of the object via `SavedObjectsSerializer.generateRawId` , it make sense yea. I would probably return undefined instead of `'N/A'` for multi-namespaces objects though in that case, or at least make sure that `undefined` is passed down to `generateRawId` for such objects.